src/ui/public/vis/editors/default/default.js

```diff
@@ -127,6 +137,32 @@ const defaultEditor = function ($rootScope, $compile) {
             catch (e) {} // eslint-disable-line no-empty
           }, true);
 
+          $scope.$watch(() => {
+            return $scope.vis.getCurrentState(false);
+          }, (newState) => {
+            // figure out what changed and copy into editor state
+            const recursiveCopyChanged = (src, org, dst) => {
+              for (const prop in src) {
+                if (typeof src[prop] === 'object') {
+                  if (!dst[prop]) {
+                    dst[prop] = _.cloneDeep(src[prop]);
+                    org[prop] = _.cloneDeep(src[prop]);
+                  } else {
+                    recursiveCopyChanged(src[prop], org[prop], dst[prop]);
```

Comment:
Why do we continue recursing under this condition (when this node is an object, and the node already exists in the editor state)? Why not update it?

Reply:
when a node is an object we need to recurse, as we want to only update specific keys.
if the object is not find in the editorState, then the whole object is copied over.